Lib/string.py

```diff
@@ -89,6 +66,25 @@ class Template(metaclass=_TemplateMetaclass):
     braceidpattern = None
     flags = _re.IGNORECASE
 
+    def __init_subclass__(cls):
+        super().__init_subclass__()
+        if 'pattern' in cls.__dict__:
+            pattern = cls.pattern
+        else:
+            pattern = r"""
+            %(delim)s(?:
+              (?P<escaped>%(delim)s) |   # Escape sequence of two delimiters
+              (?P<named>%(id)s)      |   # delimiter and a Python identifier
+              {(?P<braced>%(bid)s)}  |   # delimiter and a braced identifier
+              (?P<invalid>)              # Other ill-formed delimiter exprs
+            )
+            """ % {
```

Comment:
Too bad we can't convert this to a `.format()` string without breaking backward compatibility.

Reply:
We can. We can even convert it to f-string. I tried, but returned to the old format.

Reply:
The only problem with `.format()` and f-string is that we nee to duplicate literal braces. And the template/f-string will look strange, because braces have special meaning in regexpes and in new format.

Reply:
With f-string it would look:
```python
    def __init_subclass__(cls):
        super().__init_subclass__()
        if 'pattern' in cls.__dict__:
            pattern = cls.pattern
        else:
            delim = _re.escape(cls.delimiter)
            id = cls.idpattern
            bid = cls.braceidpattern or cls.idpattern
            pattern = fr"""
            {delim}(?:
              (?P<escaped>{delim})  |   # Escape sequence of two delimiters
              (?P<named>{id})       |   # delimiter and a Python identifier
              {{(?P<braced>{bid})}} |   # delimiter and a braced identifier
              (?P<invalid>)             # Other ill-formed delimiter exprs
            )
            """
        cls.pattern = _re.compile(pattern, cls.flags | _re.VERBOSE)
```